### ISSUE TITLE
Bug: nested struct initialization recognized as function body block

### ISSUE THREAD

**Issue by radoslav06** [uncrustify/uncrustify#4359]
Affected version: 0.79.0

Affected options:  
- `nl_after_func_body`  
- `nl_min_after_func_body`  
- probably some other related to function body

How to reproduce:  
Set `nl_min_after_func_body` to 2 (or more) to force inserting new line after function body.

Observed incorrect behavior:  
The new line is inserted within nested struct initializer.

Debug on pastebin: https://pastebin.com/ui25mYCP

Sample input code:
```C
typedef struct
{
    char data;
    struct
    {
        char innerField1;
        char innerField2;
    } innerStruct;
} ExampleStruct_t;

int main()
{
    ExampleStruct_t someStruct =
    {
        .data = 0x01U,
        .innerStruct =
        {
            .innerField1 = 1,
            .innerField2 = 2,
        } /* New line is incorrectly inserted here */
    };

    return 0;
}
```

**Comment by guy-maurel** [comment]
This seems to be the output. Where is the input? the expected output?

**Comment by radoslav06** [comment]
No, its the input. The output shall be the same as input, but instead, the new line is inserted.
Did you try to run it by yourself? Did you look into the debug file?

**Comment by guy-maurel** [comment]
I take part of [ui25mYCP](https://pastebin.com/ui25mYCP) as config file,
run uncrustify and get an output which is the same as the input

**Comment by radoslav06** [comment]
It's strange you cannot reproduce it. Here is my results from the terminal:

![image](https://github.com/user-attachments/assets/6e242787-645f-4ea7-9b78-5a33badea771)

**Comment by guy-maurel** [comment]
Sorry. I used a xyz.cpp file.
I can with .c file reproduzieren.
